### DISCUSSION TITLE
Extension to store the consumerId during the contract negotiation process

### DISCUSSION THREAD

**Discussion by boxer-code** [eclipse-edc/Connector#3837] [Q&A] [answered]
Hello everyone! I'm trying to identify each connector during the contract negotiation and store their Ids in a file. Therefore, I want to adapt the contract-negotiation extension. Every time a contract negotiation is started, I want to write the consumerId in a file. I've tried to just copy the management-api:contract-negotiation-api, adapt it and use it. But then the connector uses the api from the imported overall management-api instead of my adapted version. I also tried to copy the whole management-api but it throws weird errors. Has anyone an idea, how to exchange just that one part of the management extension or how to write an own extension which is capable of reading the ids and store them? Thanks in advance!

**Comment by jimmarino** [comment, marked as answer]
I don't understand the use case because storing participant IDs in a file will not work in most production scenarios, given that deployments are typically done in clusters. Also, participant IDs identify both counter-parties so there is no need to do this. 

The core contract negotiation module is not designed to be replaced. The proper way to integrate with these modules is typically via the eventing system. This is well documented in the code base.
